src/mbgl/layout/symbol_layout.cpp

```diff
@@ -358,6 +359,69 @@ std::array<float, 2> SymbolLayout::evaluateVariableOffset(style::SymbolAnchorTyp
+    std::optional<VariableAnchorOffsetCollection> result;
+
+    // If style specifies text-variable-anchor-offset, just return it
+    if (!textVariableAnchorOffset.isUndefined()) {
+        std::vector<AnchorOffsetPair> anchorOffsets;
+        auto variableAnchorOffset = layout->evaluate<TextVariableAnchorOffset>(zoom, feature, canonicalID);
+        if (!variableAnchorOffset.empty()) {
+            anchorOffsets.reserve(variableAnchorOffset.size());
+            // Convert offsets from EM to PX, and apply baseline shift
+            for (const auto& anchorOffset : variableAnchorOffset) {
+                std::array<float, 2> variableTextOffset = {
+                    {anchorOffset.offset[0] * util::ONE_EM, anchorOffset.offset[1] * util::ONE_EM}};
+                switch (anchorOffset.anchorType) {
+                    case SymbolAnchorType::TopRight:
+                    case SymbolAnchorType::TopLeft:
+                    case SymbolAnchorType::Top:
+                        variableTextOffset[1] -= baselineOffset;
+                        break;
+                    case SymbolAnchorType::BottomRight:
+                    case SymbolAnchorType::BottomLeft:
+                    case SymbolAnchorType::Bottom:
+                        variableTextOffset[1] += baselineOffset;
+                        break;
+                    case SymbolAnchorType::Center:
+                    case SymbolAnchorType::Left:
+                    case SymbolAnchorType::Right:
+                        break;
+                }
+
+                anchorOffsets.emplace_back(AnchorOffsetPair{anchorOffset.anchorType, variableTextOffset});
+            }
+
+            result = VariableAnchorOffsetCollection(std::move(anchorOffsets));
+        }
+    } else {
+        const std::vector<TextVariableAnchorType> variableTextAnchor = layout->evaluate<TextVariableAnchor>(
+            zoom, feature, canonicalID);
+        if (!variableTextAnchor.empty()) {
+            std::array<float, 2> variableTextOffset;
+            if (!textRadialOffset.isUndefined()) {
+                variableTextOffset = {{layout->evaluate<TextRadialOffset>(zoom, feature, canonicalID) * util::ONE_EM,
+                                       INVALID_OFFSET_VALUE}};
+            } else {
+                variableTextOffset = {{layout->evaluate<TextOffset>(zoom, feature, canonicalID)[0] * util::ONE_EM,
+                                       layout->evaluate<TextOffset>(zoom, feature, canonicalID)[1] * util::ONE_EM}};
+            }
+
+            std::vector<AnchorOffsetPair> anchorOffsets;
+            anchorOffsets.reserve(variableTextAnchor.size());
+            for (auto anchor : variableTextAnchor) {
+                auto offset = variableTextOffset;
+                offset = SymbolLayout::evaluateVariableOffset(anchor, offset);
+                anchorOffsets.emplace_back(AnchorOffsetPair{anchor, offset});
```

Comment:
This will construct an `AnchorOffsetPair` and then use the move constructor. Exactly the same as wen using `push_back`. So I would use `push_back` here.